pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go

```diff
@@ -335,15 +330,22 @@ func (n *klusterletController) sync(ctx context.Context, controllerContext facto
 	}
 	relatedResources = append(relatedResources, statuses...)
 
-	// If cluster name is empty, read cluster name from hub config secret
+	// If cluster name is empty, read cluster name from hub config secret.
+	// registration-agent generated the cluster name and set it into hub config secret.
 	if config.ClusterName == "" {
-		clusterName := hubSecret.Data["cluster-name"]
-		if clusterName != nil {
-			config.ClusterName = string(clusterName)
+		config.ClusterName, err = n.getClusterNameFromHubKubeConfigSecret(ctx, &config)
```

Comment:
if using pointer here. it does not need to return and set, right?

Reply:
no, I don't think we should deploy work agent deployment if clusterName is empty when failed to get hub-kubeconfig secret or there is no clusterName in hub-kubeconfig secret.

Reply:
I got your point.  you mean this ?
```
clusterName := n.getClusterNameFromHubKubeConfigSecret(ctx, &config)
if clusterName == nil {
 return fmt.Errorf("failed to get clusterName")
}
```

Reply:
```go
err := n.setClusterNameFromHubKubeConfigSecret(ctx, &config)
if err != nil {
  return err
}
```

Reply:
done